client/__tests__/list.test.tsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react'
+import List from '../src/pages/list';
+import { findByRole, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { setupServer, SetupServerApi } from 'msw/node';
+import { graphql } from 'msw';
+import { ANILIST_GRAPHQL_ENDPOINT } from '../src/utils/createApolloAnilist';
+import { ApolloClient, ApolloProvider, createHttpLink,  InMemoryCache, NormalizedCacheObject } from '@apollo/client';
+import router from 'next/router';
+import clientRouter from 'next/dist/client/router';
+
+import 'cross-fetch/polyfill';
+import '@testing-library/jest-dom'
+import { assert } from 'console';
+
+const FAKE_BACKEND_ENDPOINT = 'https://localhost/graphql';
+const anilist = graphql.link(ANILIST_GRAPHQL_ENDPOINT);
+
+var apolloClient: ApolloClient<NormalizedCacheObject>;
+
+const WrappedListPage: React.FC = (props) => {
+  return (
+    <ApolloProvider client={apolloClient}>
+      <List />
+    </ApolloProvider>
+  )
+}
+
+var server: SetupServerApi;
+beforeAll(() => {
+  apolloClient = new ApolloClient({
+    link: createHttpLink({ uri: FAKE_BACKEND_ENDPOINT }),
+    cache: new InMemoryCache(),
+  });
+
+  server = setupServer(
```

Comment:
I see there are two different variables `graphql` and `anilist` which I think to mock responses for queries.
I see that `anilist` has an explicit link to the anilist endpoint so I asusme `msw` is intercepting those calls to provide mock responses.
- Is `graphql` magically associated with the "default" apollo client used by the ApolloProvider? 

I also see when you mock the request, you identify the request by the query name. For example the `graphql.query('UserList'...` means intercept the query with the name `UserList` (which is defined in `graphql` directory). The `FetchAnimeInfo` query is also named. 
- So I was wondering if the query was not named would it still be possible to intercept and mock the response assuming `msw` is intercepting a query using the query name? 

If it isn't easy then it would be a good idea to always name our graphql queries in the frontend in the future as a convention.

Reply:
- the imported `graphl` scope is *not* directly associated with the default apollo client. but handlers added to that scope will apply to any graphql query regardless of url, so they'll apply to any queries that use the default apollo client. I think we could use the `graphql` scope to mock the anilist queries as well since the query names don't collide, but I'm just using it here to make it clear that we're mocking a different api
- based on reading the APIs it seems we only have the choice between mocking by query name, and providing a mock response that applies to any name/operation (not desirable, seems to be mainly for simulating error states). 

this is actually one of the main gripes I have with the msw API -- from my perspective the server wouldn't care what the query name is, so to mock that behavior I would want to be able to specify mock responses based on just the root queries or mutations. however in reality we need to know (1) the query name and (2) know the query shape so we can return data in the right shape. both of these are imo opposed to black-box testing so I wish we could do it in another way